include/eve/traits/overload/supports.hpp

```diff
@@ -0,0 +1,298 @@
+  //  Also returns an empty set of options by default
+  //====================================================================================================================
+  template<typename T, typename...> EVE_FORCEINLINE constexpr auto defaults()
+  {
+    if constexpr(requires { T::defaults(); }) return T::defaults();
+    else                                      return options{};
+  }
+
+  //====================================================================================================================
+  // Internal concept checking if an option is supported by a specification
+  //====================================================================================================================
+  template<typename Spec, typename O>
+  concept supports_option = requires(Spec sp, O const& o) { { sp[o] }; };
+
+  //====================================================================================================================
+  // Internal concept checking if an option is supported by a any specification in a given subset
+  //====================================================================================================================
+  template<typename O, typename... Specs>
+  concept is_supported = (supports_option<Specs,O> || ... );
+
+  //====================================================================================================================
+  // Internal option carrying conditional mask or conditional expressions
+  //====================================================================================================================
+  struct condition_key_t : rbr::as_keyword<condition_key_t>
+  {
+    using rbr::as_keyword<condition_key_t>::operator=;
+    template<typename T> static constexpr bool check()
+    {
+      return conditional_expr<std::remove_cvref_t<T>> || logical_value<std::remove_cvref_t<T>>;
+    }
+  };
+}
+
+namespace eve
+{
+  //====================================================================================================================
+  //! @addtogroup extensions
+  //! @{
+  //!   @var condition_key
+  //!   @brief Keyword for retrieving conditionals decorator
+  //!
+  //!   **Defined in Header**
+  //!
+  //!   @code
+  //!   #include <eve/module/core.hpp>
+  //!   @endcode
+  //!
+  //!   When a eve::decorated_callable is called with an option provided via eve::conditional, the value of the
+  //!   conditional can be retrieved using eve::condition_key
+  //!   @see eve::supports
+  //! @}
+  //====================================================================================================================
+  inline constexpr detail::condition_key_t condition_key = {};
+}
+
+namespace eve::detail
+{
+  //====================================================================================================================
+  // Intermediate type for handling chains of option like in func[opt1 = y][opt2 = x][ignore](...)
```

Comment:
we don't really write options like this in eve, looks confusing I think

Reply:
Oh yes, I forgot to change the doc here

Reply:
Fixed